src/module/iotjs_module_dns.c

```diff
@@ -15,30 +15,36 @@
 
 #include "iotjs_def.h"
 
+#include "iotjs_module_dns.h"
+
 #include "iotjs_reqwrap.h"
 #include "uv.h"
 
 
-typedef struct {
-  iotjs_reqwrap_t reqwrap;
-  uv_getaddrinfo_t req;
-} IOTJS_VALIDATED_STRUCT(iotjs_getaddrinforeqwrap_t);
-
-
 #define THIS iotjs_getaddrinforeqwrap_t* getaddrinforeqwrap
 
-void iotjs_getaddrinforeqwrap_initialize(THIS, const iotjs_jval_t* jcallback) {
+iotjs_getaddrinforeqwrap_t* iotjs_getaddrinforeqwrap_create(
+    const iotjs_jval_t* jcallback) {
+  iotjs_getaddrinforeqwrap_t* getaddrinforeqwrap =
+      IOTJS_ALLOC(iotjs_getaddrinforeqwrap_t);
   IOTJS_VALIDATED_STRUCT_CONSTRUCTOR(iotjs_getaddrinforeqwrap_t,
                                      getaddrinforeqwrap);
-  iotjs_reqwrap_initialize(&_this->reqwrap, jcallback, (uv_req_t*)&_this->req,
-                           getaddrinforeqwrap);
+  iotjs_reqwrap_initialize(&_this->reqwrap, jcallback, (uv_req_t*)&_this->req);
+  return getaddrinforeqwrap;
 }
 
 
-void iotjs_getaddrinforeqwrap_destroy(THIS) {
+static void iotjs_getaddrinforeqwrap_destroy(THIS) {
```

Comment:
What is `THIS` for? It seems not to be C, but C translated C++ code. Although I can figure out what is `THIS` by searching and thinking, for me, it needs time and hard to understand. Because I don't know the rule you translate C++ to C, and many of reader may not know it has history of traslated from C++. Could you explain what is `THIS` in comment or let it exposed directly, i.e. without macro.